compiler/hash-parser/src/operator.rs

```diff
@@ -47,121 +56,217 @@ impl Operator {
-            Operator::Shr => "shr",
-            Operator::Shl => "shl",
-            Operator::Add => "add",
-            Operator::Sub => "sub",
-            Operator::Mul => "mul",
-            Operator::Div => "div",
-            Operator::Mod => "mod",
-            Operator::As => todo!(),
+            OperatorKind::EqEq => "eq",
+            OperatorKind::NotEq => "neq",
+            OperatorKind::BitOr => "bit_or",
+            OperatorKind::Or => "orl",
+            OperatorKind::BitAnd => "bit_and",
+            OperatorKind::And => "and",
+            OperatorKind::BitXor => "bit_xor",
+            OperatorKind::Exp => "exp",
+            OperatorKind::Gt => "gt",
+            OperatorKind::GtEq => "gt_eq",
+            OperatorKind::Lt => "lt",
+            OperatorKind::LtEq => "lt_eq",
+            OperatorKind::Shr => "shr",
+            OperatorKind::Shl => "shl",
+            OperatorKind::Add => "add",
+            OperatorKind::Sub => "sub",
+            OperatorKind::Mul => "mul",
+            OperatorKind::Div => "div",
+            OperatorKind::Mod => "mod",
+            OperatorKind::As => "as",
         }
     }
 
     /// Compute the precedence for an operator
     pub(crate) fn infix_binding_power(&self) -> (u8, u8) {
         match self {
-            Operator::As => todo!(),
-            Operator::And | Operator::Or => (1, 2),
-            Operator::EqEq | Operator::NotEq => (3, 4),
-            Operator::Gt | Operator::GtEq | Operator::Lt | Operator::LtEq => (5, 6),
-            Operator::BitOr | Operator::BitAnd | Operator::BitXor => (3, 4),
-            Operator::Add | Operator::Sub => (7, 8),
-            Operator::Shr | Operator::Shl => (9, 10),
-            Operator::Mul | Operator::Div | Operator::Mod => (9, 10),
-            Operator::Exp => (11, 12),
+            OperatorKind::Or => (2, 3),
+            OperatorKind::And => (4, 5),
+            OperatorKind::EqEq | OperatorKind::NotEq => (6, 5),
+            OperatorKind::Gt | OperatorKind::GtEq | OperatorKind::Lt | OperatorKind::LtEq => (7, 8),
+            OperatorKind::BitOr | OperatorKind::BitXor => (9, 10),
+            OperatorKind::BitAnd => (11, 12),
+            OperatorKind::Shr | OperatorKind::Shl => (13, 14),
+            OperatorKind::Add | OperatorKind::Sub => (15, 16),
+            OperatorKind::Mul | OperatorKind::Div | OperatorKind::Mod => (17, 18),
+            OperatorKind::Exp => (20, 19),
+            OperatorKind::As => (21, 22),
         }
     }
 }
 
-pub enum CompoundFn {
-    Leq,
-    Geq,
-    Lt,
-    Gt,
-}
+impl From<Operator> for OperatorFn {
+    fn from(op: Operator) -> Self {
+        let Operator { kind, assignable } = op;
 
-pub enum OperatorFn {
-    Named { name: &'static str, assigning: bool },
-    LazyNamed { name: &'static str, assigning: bool },
-    Compound { name: CompoundFn, assigning: bool },
-}
+        match kind {
+            OperatorKind::As => panic!("Cannot convert 'as' into a function call."),
```

Comment:
When we have types as values, this will probably be possible :).

Reply:
indeed